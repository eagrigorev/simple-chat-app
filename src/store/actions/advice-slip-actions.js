import {
    ADVICE_SLIP_API,
    ADVICE_SLIP_SET_IDLE,
    ADVICE_SLIP_SET_LOADING,
    ADVICE_SLIP_SET_ERROR,
    ADVICE_SLIP_GENERATE_QUOTE,
} from "../../const";
export const adviceSlipSetIdle = () => ({
    type: ADVICE_SLIP_SET_IDLE,
});
export const adviceSlipSetLoading = () => ({
    type: ADVICE_SLIP_SET_LOADING,
});
export const adviceSlipSetError = () => ({
    type: ADVICE_SLIP_SET_ERROR,
});
export const adviceSlipGenerateQuote = (quote) => ({
    type: ADVICE_SLIP_GENERATE_QUOTE,
    payload: {
        quote,
    },
});
export const fetchAdviceSlipQuote = () => {
    return (dispatch, getState) => {
        dispatch(adviceSlipSetLoading());
        fetch(ADVICE_SLIP_API)
            .then((response) => {
                if (!response.ok || response.status !== 200) {
                    throw Error(`Error # ${response.status}`);
                }
                return response.json();
            })
            .then((responseJson) => {
                console.log(responseJson.slip.advice);
                dispatch(adviceSlipSetIdle());
                dispatch(adviceSlipGenerateQuote(responseJson.slip.advice));
            })
            .catch((error) => {
                console.error("error", error);
                dispatch(adviceSlipSetError());
            });
    };
};
