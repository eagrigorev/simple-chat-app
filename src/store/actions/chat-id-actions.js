import { HANDLE_CHAT_ID } from "../../const";
import firebase from "firebase";
export const handleChatId = (chatId) => ({
    type: HANDLE_CHAT_ID,
    payload: {
        chatId,
    },
});
export const fbSetChatId = (chatId) => async () => {
    firebase.database().ref("chatId").set(chatId);
};
export const fbSetChatIdThunk = (chatId) => {
    return (dispatch, getState) => {
        dispatch(fbSetChatId(chatId));
    };
};
export const fbGetChatId = () => {
    return (dispatch, getState) => {
        firebase
            .database()
            .ref("chatId")
            .on("value", (snapshot) => {
                const fbChatId = snapshot.val();
                if (fbChatId === null) {
                    dispatch(handleChatId(0));
                } else {
                    dispatch(handleChatId(fbChatId));
                }
            });
    };
};
