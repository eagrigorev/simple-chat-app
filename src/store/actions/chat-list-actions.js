import { CHAT_LIST_ADD_ITEM, CHAT_LIST_UPDATE_FROM_FB } from "../../const";
import firebase from "firebase";
export const chatListAddItem = (chatListId, chatListName) => ({
    type: CHAT_LIST_ADD_ITEM,
    payload: {
        chatListId,
        chatListName,
    },
});
export const fbChatListAddItem = (chatListId, chatListName) => async () => {
    const message = {
        chatListId: chatListId,
        chatListName: chatListName,
        chatListBot: "Bot",
    };
    firebase.database().ref("chatList").push(message);
};
export const fbChatListAddItemThunk = (chatListId, chatListName) => {
    return (dispatch, getState) => {
        dispatch(fbChatListAddItem(chatListId, chatListName));
    };
};
export const chatListUpdateFromFb = (fbChatList) => ({
    type: CHAT_LIST_UPDATE_FROM_FB,
    payload: {
        fbChatList,
    },
});
export const fbChatListFetchStoreUpdate = () => {
    return (dispatch, getState) => {
        firebase
            .database()
            .ref("chatList")
            .on("value", (snapshot) => {
                const fbChatList = [];
                snapshot.forEach((entry) => {
                    fbChatList.push(entry.val());
                });
                console.log(fbChatList);
                dispatch(chatListUpdateFromFb(fbChatList));
            });
    };
};
