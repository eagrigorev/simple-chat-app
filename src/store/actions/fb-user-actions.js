import {
    FB_USER_SET_EMAIL,
    FB_USER_SET_PASSWORD,
    FB_USER_SET_LOGGED_IN,
    FB_USER_SET_LOGGED_OUT,
} from "../../const";
import firebase from "firebase";
export const fbUserSetLoggedIn = () => ({
    type: FB_USER_SET_LOGGED_IN,
});
export const fbUserSetLoggedOut = () => ({
    type: FB_USER_SET_LOGGED_OUT,
});
export const fbUserSetEmail = (email) => ({
    type: FB_USER_SET_EMAIL,
    payload: {
        email,
    },
});
export const fbUserSetPassword = (password) => ({
    type: FB_USER_SET_PASSWORD,
    payload: {
        password,
    },
});
export const fbUserSetStatus = () => {
    return (dispatch, getState) => {
        firebase.auth().onAuthStateChanged((fbUser) => {
            if (fbUser) {
                dispatch(fbUserSetLoggedIn());
            } else {
                dispatch(fbUserSetLoggedOut());
            }
        });
    };
};
