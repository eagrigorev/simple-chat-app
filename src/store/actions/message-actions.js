import { MESSAGE_NAME_CHANGE, MESSAGE_BODY_CHANGE } from "../../const";
export const messageNameChange = (messageName) => ({
    type: MESSAGE_NAME_CHANGE,
    payload: {
        messageName,
    },
});
export const messageBodyChange = (messageBody) => ({
    type: MESSAGE_BODY_CHANGE,
    payload: {
        messageBody,
    },
});
