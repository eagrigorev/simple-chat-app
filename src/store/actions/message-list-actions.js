import firebase from "firebase";
import {
    MESSAGE_LIST_ADD_USER,
    MESSAGE_LIST_ADD_BOT,
    MESSAGE_LIST_UPDATE_FROM_FB,
} from "../../const";
export const messageListAddUser = (messageListName, messageListBody) => ({
    type: MESSAGE_LIST_ADD_USER,
    payload: {
        messageListName,
        messageListBody,
    },
});
export const messageListAddBot = (messageListName, messageListBody) => ({
    type: MESSAGE_LIST_ADD_BOT,
    payload: {
        messageListName,
        messageListBody,
    },
});
export const messageListAddBotThunk = (messageListName, messageListBody) => {
    return (dispatch, getState) => {
        setTimeout(() => {
            dispatch(messageListAddBot(messageListName, messageListBody));
        }, 3000);
    };
};
export const fbMessageListAddUser = (messageName, messageBody) => async () => {
    const message = {
        messageListName: messageName,
        messageListBody: messageBody,
    };
    firebase.database().ref("messageList").push(message);
};
export const fbMessageListAddBot = (messageName) => async () => {
    const botMessage = {
        messageListName: "Bot",
        messageListBody: `This is a bot reply to ${messageName}. That message was generated by Firebase in three seconds.`,
    };
    firebase.database().ref("messageList").push(botMessage);
};
export const fbMessageListAddUserThunk = (messageName, messageBody) => {
    return (dispatch, getState) => {
        dispatch(fbMessageListAddUser(messageName, messageBody));
    };
};
export const fbMessageListAddBotThunk = (messageName) => {
    return (dispatch, getState) => {
        setTimeout(() => {
            dispatch(fbMessageListAddBot(messageName));
        }, 3000);
    };
};
export const messageListUpdateFromFb = (fbMessageList) => ({
    type: MESSAGE_LIST_UPDATE_FROM_FB,
    payload: {
        fbMessageList,
    },
});
export const fbMessageListFetchStoreUpdate = () => {
    return (dispatch, getState) => {
        firebase
            .database()
            .ref("messageList")
            .on("value", (snapshot) => {
                const fbMessageList = [];
                snapshot.forEach((entry) => {
                    fbMessageList.push(entry.val());
                });
                console.log(fbMessageList);
                dispatch(messageListUpdateFromFb(fbMessageList));
            });
    };
};
