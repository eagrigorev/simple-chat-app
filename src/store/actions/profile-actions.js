import { PROFILE_NAME_CHANGE, PROFILE_CHECKBOX } from "../../const";
export const nameChange = (name) => ({
    type: PROFILE_NAME_CHANGE,
    payload: {
        name,
    },
});
export const checkboxChange = (isChecked) => ({
    type: PROFILE_CHECKBOX,
    payload: {
        isChecked,
    },
});
