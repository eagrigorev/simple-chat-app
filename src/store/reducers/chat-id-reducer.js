import { HANDLE_CHAT_ID } from "../../const";
const initialIdState = {
    chatId: 0,
};
export const chatIdReducer = (state = initialIdState, action) => {
    switch (action.type) {
        case HANDLE_CHAT_ID: {
            return {
                ...state,
                chatId: action.payload.chatId,
            };
        }
        default:
            return state;
    }
};
