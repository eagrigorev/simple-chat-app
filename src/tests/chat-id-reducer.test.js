import { handleChatId } from "../store/actions/chat-id-actions";
import { chatIdReducer } from "../store/reducers/chat-id-reducer";
describe("ChatId reducer state testing", () => {
    it("Returns a state with the new chatId after handleChatId action passed", () => {
        const expected = {
            chatId: 5,
        };
        const received = chatIdReducer(undefined, handleChatId(5));
        expect(received).toEqual(expected);
    });
    it("Returns the correct state after handleChatId action passed", () => {
        const expectedStore = chatIdReducer({ chatId: 0 }, handleChatId(5));
        expect(expectedStore).toMatchSnapshot();
    });
});
