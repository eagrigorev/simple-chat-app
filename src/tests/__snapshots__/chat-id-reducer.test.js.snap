// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatId reducer state testing Returns the correct state after handleChatId action passed 1`] = `
Object {
  "chatId": 5,
}
`;
